Fix maybe deep promise type

shreyas44 opened the pull request:
Removing the conditional type for null in `MaybePromiseDeep` seems to fix the below linked issue without removing any type safety.
I've tested it with the getters for `edges` and `pageInfo` with and without promises.

However, I'm not entirely sure if this breaks something else as it seems to be a utility used in mutliple places across the library.

Closes #853﻿

jasonkuhrt commented:
Is the change in this PR causing https://github.com/graphql-nexus/nexus/pull/906/checks?check_run_id=2589178499#step:6:12? Would seem so but not sure.

shreyas44 commented:
Yep, you're right, the change in the PR did cause the issue. It seems to stem from the fact that [TypeScript distributes unions when evaluating conditional types](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#distributive-conditional-types). 

I fixed that and also removed a redundant `MaybePromise` type from the return type of `resolve` as `MaybePromiseDeep` returns a `MaybePromise` itself.

jasonkuhrt commented:
Tests still seem to be failing though?
